src/screens/DiscoveryScreenV2/fragment/TopicFragment.js

```diff
@@ -229,10 +230,11 @@ const TopicFragment = ({
     });
 
   const __renderTopicItems = () => {
+    const dataToUse = isFirstTimeOpen ? initialTopics : topics;
     const followingTopics = [];
     const unfollowingTopics = [];
 
-    topics.forEach((item) => {
+    dataToUse.forEach((item) => {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This logic correctly uses `initialTopics` if it's the first time the component is opened. However, there's no check for whether `initialTopics` is null or undefined. If `initialTopics` is not provided or not loaded yet, this could lead to an error. Consider adding a null check before using `initialTopics`.
```diff
-    const dataToUse = isFirstTimeOpen ? initialTopics : topics;
+    const dataToUse = isFirstTimeOpen && initialTopics ? initialTopics : topics;
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->